pkgs/development/tools/rust/cargo-leptos/default.nix

```diff
@@ -0,0 +1,47 @@
+{ darwin
+, fetchFromGitHub
+, lib
+, openssl
+, pkg-config
+, rustPlatform
+, stdenv
+}:
+let
+  inherit (darwin.apple_sdk.frameworks)
+    CoreServices
+    Security;
+  inherit (lib) optionals;
+  inherit (stdenv) isDarwin isLinux;
+in
+rustPlatform.buildRustPackage rec {
+  pname = "cargo-leptos";
+  version = "0.1.8";
+
+  src = fetchFromGitHub {
+    owner = "leptos-rs";
+    repo = pname;
+    rev = version;
+    hash = "sha256-z4AqxvKu9E8GGMj6jNUAAWeqoE/j+6NoAEZWeNZ+1BA=";
+  };
+
+  cargoHash = "sha256-w/9W4DXbh4G5DZ8IGUz4nN3LEjHhL7HgybHqODMFzHw=";
+  nativeBuildInputs = optionals isLinux [ pkg-config ];
```

Comment:
This (and below) probably ought to be `!isDarwin`, because we'd probably want to do the same thing on e.g. BSD.

Reply:
Good callout, fixed